File: java/rna-transcription/src/main/java/RnaTranscription.java
class RnaTranscription {

    String transcribe(String dnaStrand) {
        if ( dnaStrand != null ) {
            char transcribedStrand[] = new char[ dnaStrand.length() ];
            for ( int i = 0; i < dnaStrand.length(); ++i ) {
                if ( dnaStrand.charAt(i) == 'G' ) { transcribedStrand[i] = 'C'; }
                else if ( dnaStrand.charAt(i) == 'C' ) { transcribedStrand[i] = 'G'; }
                else if ( dnaStrand.charAt(i) == 'T' ) { transcribedStrand[i] = 'A'; }
                else if ( dnaStrand.charAt(i) == 'A' ) { transcribedStrand[i] = 'U'; }
                else { throw new IllegalArgumentException( "Invalid input" ); }
            }
            return new String( transcribedStrand );
        } else {
            return null;
        }
    }

}

File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function () {};

DnaTranscriber.prototype.toRna = function (dna) {
    var rna = "";
    for ( var i = 0; i < dna.length; ++i ) {
        if ( dna.charAt(i) == 'C' ) { rna += 'G'; }
        else if ( dna.charAt(i) == 'G' ) { rna += 'C'; }
        else if ( dna.charAt(i) == 'A' ) { rna += 'U'; }
        else if ( dna.charAt(i) == 'T' ) { rna += 'A'; }
        else { throw new Error( "Invalid input" ); }
    }
    return rna;
};

module.exports = DnaTranscriber